### PR TITLE
Add positioning helpers for trashcan and zoom controls

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

n/a

### Proposed Changes

- Adds package-level helpers for positioning trashcan and zoom controls (this would cause bugs for anything positioned above the zoom controls).
- Changes logic for adding padding between trashcan/zoom controls and edge of the workspace to only add `scrollbarThickness` if there are scrollbars between the ui elements and the edge of the workspace. 
- Corrects bug where bounding box was incorrect when there are zoom controls without zoom reset.

#### Behavior Before Change
- Zoom controls reported same height when it has zoom reset control and without.
image
- Extra padding between the trashcan and zoom controls when the toolbox is placed at the end.
image

#### Behavior After Change
- Zoom controls does not include zoom reset control height if it doesn't exist.
- Symmetrical positioning of trashcan and zoom controls.
image

### Reason for Changes

Core reuse, readability, and fixing minor positioning bugs

### Test Coverage
Manually tested on multi_playground and playground (resizing and comparing to master).

### Additional Information

Based off of #4782

### CODE REVIEW

File: core/positionable_helpers.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+};
+
+/**
+ * Returns a rectangle representing reasonable position for where to place a ui
+ * element of the specified size given the restraints and locations of the
+ * scrollbars. This method does not take into account any already placed ui
+ * elements.
+ * @param {!Blockly.uiPosition.Position} position The starting
+ *    horizontal and vertical position.
+ * @param {!Blockly.utils.Size} size the size of the ui element so get a start
```
Comment (alschmiedt) [inline core/positionable_helpers.js:70]: nit: **T**he size of the ui element **to** get a start position for.

File: core/positionable_helpers.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+ *    position for.
+ * @param {number} horizontalPadding The horizontal padding to use. This value
+ *    is ignored for center horizontal positioning.
+ * @param {number} verticalPadding The vertical padding to use. This value
```
Comment (alschmiedt) [inline core/positionable_helpers.js:74]: Is the comment about center horizontal and vertical padding still valid? It looks like we removed the option to center positionables.

**Review by alschmiedt — APPROVED** [review]
two nits then lgtm

File: core/positionable_helpers.js (hunk 2 above)
Comment (moniika) [reply core/positionable_helpers.js]: Good catch. Yeah, it's not relevant anymore.

**Pull request merged**